etc/docker/test/matrix_nightly.yml

```diff
@@ -0,0 +1,48 @@
+dists:
+  - id: centos7
+    allow:
+      python:
+        - "2.7"
+        - "3.6"
+        - "3.7"
+  - id: fedora32
+    allow:
+      python:
+        - "3.8"
+  - id: fedora33
+    allow:
+      python:
+        - "3.9"
+python:
+  - id: "2.7"
+    allow:
+      suites:
+        - client
+        - client_syntax
+  - "3.6"
+  - "3.7"
+  - "3.8"
+  - "3.9"
+suites:
+  - id: syntax
+    SYNTAX_REPORT: 1
+    RUN_HTTPD: false
+  - id: docs
+    RUN_HTTPD: false
+  - id: client_syntax
+    RUN_HTTPD: false
+    allow:
+      python: "2.7"
+  - id: client
+    RDBMS: sqlite
+  - id: all
+    RDBMS:
+      - oracle
+      - mysql8
+      - postgres9
+      - postgres12
```

Comment:
Do we want to support `postgres14` in the nightly builds? @bari12

Reply:
my patch sets the dev default to psql14, so should be added here as well

Reply:
As discussed in marios PR https://github.com/rucio/rucio/pull/5324#discussion_r820775921
Let's just do postgres14, remove 9 and 12.